### PR TITLE
feat: support magic match completion for tuple

### PR DESCRIPTION
**Pull request by stormckey**
According to #9202, we now support magic match for tuple!

Here is how it looks like:

image

image

Note:
- I introduce a new enum `Member` to represent either a enum or other type in the tuple.
- Maybe we could consider using the same marker for all the `_member2` placeholder so that user can edit them simultaneously.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
!!!jar

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala [hunk 1]
```diff
@@ -36,78 +36,164 @@ object MagicMatchCompleter {
...
+    */
+  private def mkTupleMatchBody(members: List[List[Member]]): String = {
     val sb = new StringBuilder
-    var z = 1
-    for ((sym, cas) <- cases.toList.sortBy(_._1.loc)) {
-      val (lhs, rhs, newZ) = createCase(sym, cas, z)
-      val paddedLhs = padLhs(lhs, maxCaseLength)
-      sb.append(s"    case $paddedLhs => $rhs\n")
-      z = newZ
+    val maxCaseLength = members.map(getMemberLength).max
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala:117]: Do we know this .max cannot cause an exception?

**Review by magnus-madsen — COMMENTED** [review]
The code seems to work well, but I am struggling a bit to understand it. 

How do you think we can document it better?

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala (hunk 1 above)
Comment (stormckey) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala:117]: The .max will not cause an exception as long as the length of the tuple is not zero.

Is it possible for us to create an empty tuple in flix?

**Comment by stormckey** [comment]
I've changed part of the name to make the code easier to understand.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala]: No. So that should be good.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**